integration/sawtooth_integration/tests/test_identity.py

```diff
@@ -0,0 +1,399 @@
+            permit_keys=[self.alice.public_key, self.dave.public_key],
+            deny_keys=[self.bob.public_key])
+
+        self.assert_able_to_send(self.alice, Families.INTKEY)
+
+        for t, f in [(self.bob, Families.INTKEY), (self.chuck, Families.INTKEY),
+                     (self.mallory, Families.INTKEY), (self.dave, Families.INTKEY)]:
+            self.assert_not_able_to_send(t, f)
+
+    def assert_able_to_send(self, transactor, family):
+        transactor.send(family)
+
+    def assert_able_to_send_batch(self, transactor, family, transactions):
+        transactor.send(family_name=family, transactions=transactions)
+
+    def assert_not_able_to_send(self, transactor, family):
+        with self.assertRaises(Exception):
+            transactor.send(family)
+
+    def assert_not_able_to_send_batch(self, transactor, family, txns):
+        with self.assertRaises(Exception):
+            transactor.send(family, txns)
+
+
+INTKEY_NAMESPACE = MessageFactory.sha512('intkey'.encode())[:6]
+XO_NAMESPACE = MessageFactory.sha512('xo'.encode())[:6]
+IDENTITY_NAMESPACE = '00001d'
+
+ROLE_PREFIX = '01'
+POLICY_PREFIX = '00'
+
+
+class Families(enum.Enum):
+    INTKEY = 1
+    XO = 2
+    IDENTITY = 3
+
+
+FamilyConfig = {
+    Families.INTKEY: {
+        'encoding': 'application/cbor',
+        'family_name': 'intkey',
+        'family_version': '1.0',
+        'namespace': MessageFactory.sha256('intkey'.encode())[:6]
+    },
+    Families.XO: {
+        'encoding': 'csv-utf8',
+        'family_name': 'xo',
+        'family_version': '1.0',
+        'namespace': MessageFactory.sha256('xo'.encode())[:6]
+    },
+    Families.IDENTITY: {
+        'encoding': 'application/protobuf',
+        'family_name': 'identity',
+        'family_version': '1.0',
+        'namespace': '00001d'
+    }
+}
+
+
+def INTKEY_PAYLOAD(unique_value):
+    return {'Verb': 'set', 'Name': unique_value, 'Value': 1000}
+
+
+def INTKEY_ADDRESS(unique_value):
+    return INTKEY_NAMESPACE + MessageFactory.sha512(unique_value.encode())[-64:]
+
+
+def XO_PAYLOAD(unique_value):
+    return "{},{},{}".format(unique_value, 'create', '').encode('utf-8')
+
+
+def XO_ENCODER(contents):
+    return contents
+
+
+def XO_ADDRESS(unique_value):
+    return XO_NAMESPACE + MessageFactory.sha512(unique_value.encode())[:64]
+
+
+TransactionEncoder = {
+    Families.INTKEY: {
+        'encoder': cbor.dumps,
+        'payload_func': INTKEY_PAYLOAD,
+        'address_func': INTKEY_ADDRESS
+    },
+    Families.XO: {
+        'encoder': XO_ENCODER,
+        'payload_func': XO_PAYLOAD,
+        'address_func': XO_ADDRESS
+    }
+}
+
+
+class Transactor(object):
+
+    def __init__(self, name, rest_endpoint):
+        """
+        Args:
+            name (str): An identifier for this Transactor
+            rest_endpoint (str): The rest api that this Transactor will
+                communicate with.
+        """
+
+        self.name = name
+        self._rest_endpoint = rest_endpoint \
```

Comment:
`wait_for_rest_apis` incorrectly expects its argument to not have `http://` and adds it. This should be changed, so test logic shouldn't be built around that behavior. The pattern I've been using is to have to variables, `REST_API = rest-api:8080` and `URL = 'http://' + REST_API`, with `REST_API` being used for `wait_for_rest_apis` and `URL` being used for everything else.